docs: add `ignite chain debug` documentation and tutorial (#3272)
ignite/cli: docs/docs/kb/13-debug.md

Once all breakpoints are setted resume blockchain execution using the
[continue](https://github.com/go-delve/delve/blob/master/Documentation/cli/README.md#continue)
(alias `c`) command:

```
(dlv) continue
```

The debugger will launch the shell and stop blockchain execution again when a
breakpoint is triggered.

Within the debugger shell use the `quit` (alias `q`) or `exit` commands to stop
the blockchain app and exit the debugger.

## Debug Server

A debug server can optionally be started in cases where the default terminal
client is not desirable. When the server starts it first runs the blockchain
app, attaches to it and finally waits for a client connection. It accepts both
JSON-RPC or DAP client connections.

To start a debug server use the following flag:

```bash
ignite chain debug --server
```

To start a debug server with a custom address use the following flags:

```bash
ignite chain debug --server --server-address 127.0.0.1:30500
```

The debug server stops automatically when the client connection is closed.

## Debugging Clients

### Gdlv: Multiplatform Delve UI

[Gdlv](https://github.com/aarzilli/gdlv) is a graphical frontend to Delve for
Linux, Windows and macOS.

> Using it as debugging client is straight forward as it doesn't require any

Reviewer: ```suggestion
Using it as debugging client is straightforward as it doesn't require any
```

configuration. Once the debug server is running and listening for client
requests, for example at the default `tcp://127.0.0.1:30500` address, connect to

> it by running:

Reviewer: I think the "for example at the default ... address" part should be moved earlier in the DEBUG SERVER section.  Then this sentence can be simplified.

Author: Changeset bddd289

```bash
gdlv connect 127.0.0.1:30500
```

Setting breakpoints and continuing execution is done in the same way as Delve,
by using the `break` and `continue` commands.

### Visual Studio Code

Using [Visual Studio Code](https://code.visualstudio.com/) as debugging client

> requires an initial configuration to make it connect to the debug server.

Reviewer: ```suggestion
requires an initial configuration to allow it to connect to the debug server.
```

extension is installed.

VS Code debugging is configured using the `launch.json` file which is usually
located inside the `.vscode` folder in your workspace.

You can use the following launch configuration to set up VS Code as debugging 
client:

```json title=launch.json
{
    "version": "0.2.0",
    "configurations": [
        {
            "name": "Connect to Debug Server",
            "type": "go",
            "request": "attach",
            "mode": "remote",
            "remotePath": "${workspaceFolder}",
            "port": 30500,
            "host": "127.0.0.1"
        }
    ]
}
```

Alternatively it's possible to create a custom `launch.json` file from the "Run
and Debug" panel. When prompted choose the Go debugger option labeled "Go: 
Connect to Server" and enter the debug host address and then the port number.

## Example: Debugging a Blockchain App

In this short example we will be using Ignite CLI to create a new blockchain and
a query to be able to trigger a debugging breakpoint when the query is called.

To create a new blockchain and a query run:

```bash
ignite scaffold chain hello
ignite scaffold query say-hello name --response name
```

> The next step initializes the blobkchain's data directory and compiles a debug

Reviewer: ```suggestion
The next step initializes the blockchain's data directory and compiles a debug
```
